### DISCUSSION TITLE
Python 虛擬環境

### DISCUSSION THREAD

**Discussion by lx3211** [1101-GS4719-C/Python-Community-GS4719-C#457] [General]
隨著日後專案變多，每個專案需要的套件都不一樣
因此虛擬環境virtualenv就是輔助分開套件們的好工具
讓你的每個開發環境都乾乾淨淨的

- **首先先下載virtualenv，在cmd輸入下列指令**
`pip install virtualenv`

- **建立虛擬環境**
`virtualenv --python=/opt/python-3.6/bin/python venv  #指定python版本`
`python3 -m venv env_name #在哪裏執行這個指令 虛擬環境的資料夾就會建在哪裡`

- **使用虛擬環境**
- Linux/MacOS
`source ./env_name/bin/activate #要確定資料夾在哪cd過去`
- Windows
`.\env_name\Scripts\activate.bat `
&emsp;<img width="262" alt="截圖 2022-03-10 12 36 08" src="https://user-images.githubusercontent.com/96198440/157590391-710c79e6-e411-463a-a9d0-95fb04345a07.png">
&emsp;成功啟動會看到命令行前多了(env_name)的字樣

- 之後要下載套件就先啟動虛擬環境再pip 如此一來就可以把他們都分割了(´▽｀)

- **關閉虛擬環境**
`deactivate`

因為我使用的是MacOS所以Windows的操作我是網路上找來的
我不確定是否能操作
[附上windows參考資料](https://ithelp.ithome.com.tw/articles/10199980)

**Comment by ThomasMeng0515** [comment]
利用虛擬環境開發環境真的會乾淨很多，感謝分享~

**Comment by JSHT** [comment]
windows 上使用 powershell 輸入 `.\env_name\Scripts\activate` 就可以啟動虛擬環境
如果是使用較新版本的 python(EX. 3.10.2)，創建虛擬環境時可能會遇到一些錯誤。可以改輸入 `py -3 -m venv env_name`。
['virtualenv' is not recognized as an internal or external command](https://stackoverflow.com/questions/35950740/virtualenv-is-not-recognized-as-an-internal-or-external-command-operable-prog)

**Comment by ghost** [comment]
想學有關anaconda的環境建置我也推薦這個人的教學
他講的都還算詳細
https://www.youtube.com/watch?v=qLjw_EtqmCs&list=PL0e6zhuvS3U4HHxdKbzB4KciI9--TtvuG&index=3&t=1677s

**Comment by minniewei** [comment]
之前有用過python的虛擬環境真的非常好用

**Comment by minniewei** [comment]
之前有用過python的虛擬環境真的非常好用

**Comment by Rosancl** [comment]
感謝分享，用虛擬環境真的會方便好用很多

**Comment by yucheng1028** [comment]
用虛擬機就可以體驗別的作業系統了

**Comment by Lynn12378** [comment]
虛擬環境還可以檢查在建立網站所需的套件，寫入requiremets.txt

**Comment by Jimmy911218** [comment]
謝謝分享虛擬環境確實很乾淨，能在上面作事方便
